libs/devices/harmony/feature/src/lib/hooks/use-harmony-file-list.query.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { useQuery } from "@tanstack/react-query"
+import { HarmonyDirectory, Harmony } from "devices/harmony/models"
+import { harmonyQueryKeys } from "./harmony-query-keys"
+import { getHarmonyFileList } from "../api/get-harmony-file-list"
+
+const queryFn = async (directory: HarmonyDirectory, device?: Harmony) => {
+  if (!device) {
+    throw new Error("No device provided for useHarmonyFileListQuery")
+  }
+
+  const response = await getHarmonyFileList(directory, device)
+
+  if (!response.ok) {
+    throw response.status
+  }
+
+  return response.body
+}
+
+export const useHarmonyFileListQuery = (
+  directory: HarmonyDirectory,
+  device?: Harmony
+) => {
+  return useQuery({
+    queryKey: useHarmonyFileListQuery.queryKey(`${device?.path}+${directory}`),
```

Comment:
For devices, the query key should have the device path part clearly separated, so it's in the second position. This makes it easier to perform batch operations on all queries from a given device.

For instance, the key `["devices", "ABC123456+path/to/some/directory"]` won't work if we want to take action after disconnecting the device:

```TS
queryClient.removeQueries({ queryKey: ["devices", "ABC123456"] })
```